src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -165,11 +166,66 @@ osexperimental_handle_live_fs (RPMOSTreeOSExperimental *interface,
   return TRUE;
 }
 
+static gboolean
+osexperimental_handle_search (RPMOSTreeOSExperimental *interface,
+                              GDBusMethodInvocation *invocation,
+                              const gchar * const *queries)
+{
+  g_autoptr(GCancellable) cancellable = g_cancellable_new ();
+  GError **error = NULL;
+  OstreeSysroot *sysroot = rpmostreed_sysroot_get_root (rpmostreed_sysroot_get ());
+  OstreeDeployment *booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  const char *osname = ostree_deployment_get_osname (booted_deployment);
+
+  g_autoptr(OstreeDeployment) cfg_merge_deployment =
+    ostree_sysroot_get_merge_deployment (sysroot, osname);
+  g_autoptr(OstreeDeployment) origin_merge_deployment =
+    rpmostree_syscore_get_origin_merge_deployment (sysroot, osname);
+
+  /* but set the source root to be the origin merge deployment's so we pick up releasever */
+  g_autofree char *origin_deployment_root =
+    rpmostree_get_deployment_root (sysroot, origin_merge_deployment);
+
+  OstreeRepo *repo = ostree_sysroot_repo (sysroot);
+  g_autoptr(RpmOstreeContext) ctx = rpmostree_context_new_client (repo);
+  rpmostree_context_set_dnf_caching (ctx, RPMOSTREE_CONTEXT_DNF_CACHE_FOREVER);
+
+  /* We could bypass rpmostree_context_setup() here and call dnf_context_setup() ourselves
+   * since we're not actually going to perform any installation. Though it does provide us
+   * with the right semantics for install/source_root. */
+  if (!rpmostree_context_setup (ctx, NULL, origin_deployment_root, cancellable, error))
+    return FALSE;
+  /* point libdnf to our repos dir */
+  rpmostree_context_configure_from_deployment (ctx, sysroot, cfg_merge_deployment);
+
+  if (!rpmostree_context_download_metadata (ctx, DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_RPMDB,
+                                            cancellable, error))
+    return FALSE;
+
+  DnfSack *sack = dnf_context_get_sack (rpmostree_context_get_dnf (ctx));
+  g_autoptr(GPtrArray) urls = g_ptr_array_new();
+
+  for (unsigned int i = 0; queries[i] != NULL; ++i)
+    {
+      auto query = static_cast<const char*> (queries[i]);
+      g_autoptr(GPtrArray) pkglist = rpmostree_get_matching_packages (sack, query);
+      if (!pkglist)
+        return FALSE;
```

Comment:
Something like

```suggestion
      if (!pkglist || pkglist->len == 0)
        return glnx_throw (error, "No matching packages found for query '%s'", query);
```

In practice, I think `!pkglist` is never going to happen and it might be more appropriate as an assert instead, but meh...